common/app/model/trails.scala

```diff
@@ -90,7 +93,13 @@ final case class Trail (
   lazy val showByline: Boolean = tags.isComment
 
   lazy val shouldHidePublicationDate: Boolean = {
-    commercial.isAdvertisementFeature && webPublicationDate.isOlderThan(2.weeks)
+    lazy val isPaidContentInDfp = staticBadgesSwitch.isSwitchedOff && commercial.isAdvertisementFeature
+    lazy val isPaidContentInCapi =
+      staticBadgesSwitch.isSwitchedOn && BrandHunter.findTrailBranding(this, defaultEdition).exists {
```

Comment:
Just thinking here. Might be worth abstracting this expression in something like `BrandHunter.isPaidContent(this, defaultEdition)`?